src/packet/sym_encrypted_integrity_protected.js

```diff
@@ -73,82 +74,57 @@ SymEncryptedIntegrityProtected.prototype.read = function (bytes) {
-
-    } else { // asm.js fallback
-      this.encrypted = asmCrypto.AES_CFB.encrypt(util.concatUint8Array([prefix, tohash]), key);
-    }
-
+    this.encrypted = aesEncrypt(sessionKeyAlgorithm, prefix, tohash, key);
   } else {
-    this.encrypted = crypto.cfb.encrypt(prefixrandom, sessionKeyAlgorithm, tohash, key, false)
-      .subarray(0, prefix.length + tohash.length);
+    this.encrypted = crypto.cfb.encrypt(prefixrandom, sessionKeyAlgorithm, tohash, key, false);
+    this.encrypted = this.encrypted.subarray(0, prefix.length + tohash.length);
   }
 
   return Promise.resolve();
 };
 
 /**
- * Decrypts the encrypted data contained in this object read_packet must
- * have been called before
- *
- * @param {module:enums.symmetric} sessionKeyAlgorithm
- *            The selected symmetric encryption algorithm to be used
- * @param {Uint8Array} key The key of cipher blocksize length to be used
- * @return {String} The decrypted data of this packet
+ * Decrypts the encrypted data contained in the packet.
+ * @param  {String} sessionKeyAlgorithm   The selected symmetric encryption algorithm to be used e.g. 'aes128'
+ * @param  {Uint8Array} key               The key of cipher blocksize length to be used
+ * @return {Promise}
  */
 SymEncryptedIntegrityProtected.prototype.decrypt = function (sessionKeyAlgorithm, key) {
-  var decrypted;
-
+  let decrypted;
   if(sessionKeyAlgorithm.substr(0,3) === 'aes') {  // AES optimizations. Native code for node, asmCrypto for browser.
-    var blockSize = crypto.cipher[sessionKeyAlgorithm].blockSize;
-
-    if(nodeCrypto) { // Node crypto library. Only loaded if config.use_native === true
-      var decipherObj = new nodeCrypto.createDecipheriv('aes-' + sessionKeyAlgorithm.substr(3,3) + '-cfb',
-        new Buffer(key), new Buffer(new Uint8Array(blockSize)));
-      decrypted = new Uint8Array(decipherObj.update(new Buffer(this.encrypted)));
-
-    } else { // asm.js fallback
-      decrypted = asmCrypto.AES_CFB.decrypt(this.encrypted, key);
-    }
-
-    // Remove random prefix
-    decrypted = decrypted.subarray(blockSize + 2, decrypted.length);
-
+    decrypted = aesDecrypt(sessionKeyAlgorithm, this.encrypted, key);
   } else {
     decrypted = crypto.cfb.decrypt(sessionKeyAlgorithm, key, this.encrypted, false);
   }
 
   // there must be a modification detection code packet as the
   // last packet and everything gets hashed except the hash itself
-  this.hash = util.Uint8Array2str(crypto.hash.sha1(util.concatUint8Array([crypto.cfb.mdc(sessionKeyAlgorithm, key, this.encrypted),
-    decrypted.subarray(0, decrypted.length - 20)])));
-
-  var mdc = util.Uint8Array2str(decrypted.subarray(decrypted.length - 20, decrypted.length));
+  const prefix = crypto.cfb.mdc(sessionKeyAlgorithm, key, this.encrypted);
+  const bytes = decrypted.subarray(0, decrypted.length - 20);
+  const tohash = util.concatUint8Array([prefix, bytes]);
+  this.hash = util.Uint8Array2str(crypto.hash.sha1(tohash));
```

Comment:
@bartbutler this part was heavily inlined and I'm not sure there were performance reasons for it? I made it a bit more readable. I hope this doesn't affect performance.